source/Renderer/shaders/ibl.glsl

```diff
@@ -17,51 +17,48 @@ vec4 getSheenSample(vec3 reflection, float lod)
     return textureLod(u_CharlieEnvSampler, u_EnvRotation * reflection, lod) * u_EnvIntensity;
 }
 
-
-vec3 getIBLRadianceGGX(vec3 n, vec3 v, float roughness, vec3 F0, float specularWeight)
+vec3 getIBLGGXFresnel(vec3 n, vec3 v, vec3 F0, float roughness, float specularWeight)
 {
+    // see https://bruop.github.io/ibl/#single_scattering_results at Single Scattering Results
+    // Roughness dependent fresnel, from Fdez-Aguera
     float NdotV = clampedDot(n, v);
-    float lod = roughness * float(u_MipCount - 1);
-    vec3 reflection = normalize(reflect(-v, n));
-
     vec2 brdfSamplePoint = clamp(vec2(NdotV, roughness), vec2(0.0, 0.0), vec2(1.0, 1.0));
     vec2 f_ab = texture(u_GGXLUT, brdfSamplePoint).rg;
-    vec4 specularSample = getSpecularSample(reflection, lod);
-
-    vec3 specularLight = specularSample.rgb;
+    vec3 Fr = max(vec3(1.0 - roughness), F0) - F0;
+    vec3 k_S = F0 + Fr * pow(1.0 - NdotV, 5.0);
+    vec3 FssEss = specularWeight * (k_S * f_ab.x + f_ab.y);
+    return FssEss;
+}
 
-    // see https://bruop.github.io/ibl/#single_scattering_results at Single Scattering Results
-    // Roughness dependent fresnel, from Fdez-Aguera
+vec3 getIBLLambertianFresnel(vec3 n, vec3 v, float roughness, vec3 diffuseColor, vec3 F0, float specularWeight) 
+{
+    float NdotV = clampedDot(n, v);
+    vec2 brdfSamplePoint = clamp(vec2(NdotV, roughness), vec2(0.0, 0.0), vec2(1.0, 1.0));
+    vec2 f_ab = texture(u_GGXLUT, brdfSamplePoint).rg;
     vec3 Fr = max(vec3(1.0 - roughness), F0) - F0;
     vec3 k_S = F0 + Fr * pow(1.0 - NdotV, 5.0);
-    vec3 FssEss = k_S * f_ab.x + f_ab.y;
+    vec3 FssEss = specularWeight * (k_S * f_ab.x + f_ab.y); // <--- GGX / specular light contribution (scale it down if the specularWeight is low)
 
-    return specularWeight * specularLight * FssEss;
+    // Multiple scattering, from Fdez-Aguera
+    float Ems = (1.0 - (f_ab.x + f_ab.y));
+    vec3 F_avg = specularWeight * (F0 + (1.0 - F0) / 21.0);
+    vec3 FmsEms = Ems * FssEss * F_avg / (1.0 - F_avg * Ems);
+    vec3 k_D = diffuseColor * (1.0 - FssEss + FmsEms); // we use +FmsEms as indicated by the formula in the blog post (might be a typo in the implementation)
+    return (FmsEms + k_D);
```

Comment:
I think the `FmsEms` needs to go to the `getIBLGGXFresnel` function. Right now, the code seems to do both, the energy compensation for the GXX multi-scatter as well as the diffuse component scaling. This issue went unnoticed so far because metal and dielectric contributions were added anyways. Now, that the result is a [mix of the contributions based on the metallic value](https://github.com/KhronosGroup/glTF-Sample-Viewer/pull/556/files#diff-1e994bd463e43f61e373a3f86bd524d77eca7c274beca9496e89fce9b75f9058R200), the multi-scatter contribution for rough metals is lost.
```suggestion
    return k_D;
```